src/main/java/org/folio/rest/impl/PutOrderLineByIdHelper.java

```diff
@@ -205,18 +207,25 @@ public boolean isUpdateSuccessful() {
       .thenAccept(body -> {
         PieceCollection pieces = body.mapTo(PieceCollection.class);
 
-        // Extract item Id's which already associated with piece records
-        List<String> existingItemIds = pieces.getPieces()
-            .stream()
-            .map(Piece::getItemId)
-            .filter(StringUtils::isNotEmpty)
-            .collect(Collectors.toList());
-
-        // Create piece records for item id's which do not have piece records yet
-        itemIds.stream()
-               .filter(id -> !existingItemIds.contains(id))
-               .forEach(itemId -> futuresList.add(createPiece(poLineId, itemId)));
-
+        if(itemIds!=null) {
+          // Extract item Id's which are already associated with piece records
+          List<String> existingItemIds = pieces.getPieces()
+																	             .stream()
+																	             .map(Piece::getItemId)
+																	             .filter(StringUtils::isNotEmpty)
+																	             .collect(Collectors.toList());
+
+          // Create piece records for item id's which do not have piece records yet     
+	        itemIds.stream()
+	               .filter(id -> !existingItemIds.contains(id))
+	               .forEach(itemId -> futuresList.add(createPiece(poLineId, itemId)));
```

Comment:
With the latest updates one case is missed. 
Let's consider PO Line of "P/E Mix" format with 3 Physical resources and 2 Electronic resources with `create_inventory` set to `false`.
`itemIds` will not be `null` containing 3 elements. 2 piece records for Electronic resources will not be created.

Reply:
When `create_inventory` is set to `false`, `updateInventory` will not be called so `itemIds` will be `null`. Is that correct?

Reply:
Looks like `po_listed_print_monograph_pe_mix.json` created to cover this case but the physical quantity is zero in both locations. Once I modified the second location specifying `"quantity": 3`, the test has failed

![image](https://user-images.githubusercontent.com/43410167/52775783-872f0280-3051-11e9-93f8-9197453beca7.png)